src/common/tls_session.h

```diff
@@ -0,0 +1,97 @@
+/*
+   Copyright 2023 Leil Storage OÜ
+
+   This file is part of SaunaFS.
+
+   SaunaFS is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, version 3.
+
+   SaunaFS is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with SaunaFS  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "common/platform.h"
+
+#include <memory>
+#include <string>
+
+#include <openssl/err.h>
+#include <openssl/ssl.h>
+
+#include "common/exception.h"
+
+SAUNAFS_CREATE_EXCEPTION_CLASS(OpenSslException, Exception);
+
+inline std::string opensslErrorString(int sslErrorCode) {
+	switch (sslErrorCode) {
+	case SSL_ERROR_WANT_READ:
+		return "WANT_READ";
+	case SSL_ERROR_WANT_WRITE:
+		return "WANT_WRITE";
+	case SSL_ERROR_ZERO_RETURN:
+		return "PEER_CLOSED";
+	case SSL_ERROR_SYSCALL:
+		return "SYSCALL_ERROR";
+	case SSL_ERROR_SSL: {
+		unsigned long err = ERR_get_error();
+		if (err) {
+			char buf[256];
+			ERR_error_string_n(err, buf, sizeof(buf));
+			return std::string("SSL_PROTOCOL_ERROR: ") + buf;
+		}
+		return "SSL_PROTOCOL_ERROR";
+	}
+	default:
+		return "UNKNOWN_ERROR";
+	}
+}
+
+class TlsSession {
+public:
+	static const std::string kNoFile;
```

Comment:
Consider:

```suggestion
	static constexpr std::string_view kNoFile = "/unused file/";
```

This will remove the out-of-line definition in the .cc file. The comparisons should work, but the calls to cfg_getstring will need to use std::string(kNoFile), which should not be a problem.